prm_parameter_writer.cpp

```diff
@@ -0,0 +1,91 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by Rene Gassmoeller
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+#include <QtGui>
+
+#include "prm_parameter_writer.h"
+
+
+namespace dealii
+{
+  namespace ParameterGui
+  {
+    PRMParameterWriter::PRMParameterWriter(QTreeWidget *tree_widget)
+                      : tree_widget(tree_widget)
+    {
+    }
+
+
+
+    bool PRMParameterWriter::write_prm_file(QIODevice *device)
+    {
+										// loop over the elements
+      for (int i = 0; i < tree_widget->topLevelItemCount(); ++i)
+        {
+        const QString item_string = write_item(tree_widget->topLevelItem(i),0);
+        device->write(item_string.toAscii());
+        }
+
+      return true;
+    }
+
+
+
+    QString PRMParameterWriter::write_item(const QTreeWidgetItem *item,
+                                           const unsigned int indentation_level)
+    {
+      QString item_string;
+
+      if (item->childCount() == 0)						// if the "value"-entry of this item is not empty
+        {									// we have a parameter
+          bool non_default_value;
+
+          if (item->text(5).contains("Double"))
```

Comment:
is ``text(5)`` the datatype? Can you use ``startsWith`` instead? Otherwise it might match a ``Selection [ ... double`` or something.